controlplane/src/core/repositories/analytics/MetricsDashboardRepository.ts

```diff
@@ -0,0 +1,364 @@
+   */
+  public async getRequestRateMetrics({ range = 24, endDate = getUtcUnixTime(), params }: GetMetricsProps) {
+    const prange = '15m';
+
+    // get median of requests in last [range] hours
+    const median = await this.client.query({
+      query: `quantile(0.5, sum(rate(cosmo_router_http_requests_total{${this.getQueryLabels(
+        params,
+      )}}[${range}h])) * 60)`,
+    });
+
+    const prevMedian = await this.client.query({
+      query: `quantile(0.5, sum(rate(cosmo_router_http_requests_total{${this.getQueryLabels(
+        params,
+      )}}[${range}h] offset ${range}h)) * 60)`,
+    });
+
+    // get top 5 operations in last [range] hours
+    const top5 = await this.client.query({
+      query: `topk(5, sum by (wg_operation_name) (rate(cosmo_router_http_requests_total{${this.getQueryLabels(
+        params,
+      )}}[${range}h])) * 60)`,
+    });
+
+    // get requests in last [range] hours in series of [step]
+    const result = await this.client.queryRange({
+      query: `sum (rate(cosmo_router_http_requests_total{${this.getQueryLabels(params)}}[${prange}]) * 60)`,
+      ...this.getRangeQueryProps({
+        range,
+        endDate,
+      }),
+    });
+
+    const prevResult = await this.client.queryRange({
+      query: `sum (rate(cosmo_router_http_requests_total{${this.getQueryLabels(params)}}[${prange}]) * 60)`,
+      ...this.getRangeQueryProps({
+        range,
+        endDate: endDate - range * 60 * 60 * 1000,
+      }),
+    });
+
+    return {
+      value: median.data.result[0]?.value[1],
+      previousValue: prevMedian.data.result[0]?.value[1],
+      top: top5.data.result.map((v: any) => ({
+        name: v.metric.wg_operation_name,
+        value: v.value[1],
+      })),
+      series: this.mapSeries(endDate, range, result.data.result[0].values, prevResult.data.result[0]?.values),
+    };
+  }
+
+  /**
+   * Get latency metrics
+   */
+  public async getLatencyMetrics({ range = 24, endDate = getUtcUnixTime(), params }: GetMetricsProps) {
+    const prange = '15m';
+
+    // get p95 of requests in last [range] hours
+    const latencyP95 = await this.client.query({
+      query: `histogram_quantile(0.95, sum(rate(cosmo_router_http_request_duration_milliseconds_bucket{${this.getQueryLabels(
+        params,
+      )}}[${range}h])) by (le))`,
+    });
+
+    const prevLatencyP95 = await this.client.query({
+      query: `histogram_quantile(0.95, sum(rate(cosmo_router_http_request_duration_milliseconds_bucket{${this.getQueryLabels(
+        params,
+      )}}[${range}h] offset ${range}h)) by (le))`,
+    });
+
+    // get top 5 operations in last [range] hours
+    const top5 = await this.client.query({
+      query: `topk(5, sum by (wg_operation_name) (rate(cosmo_router_http_request_duration_milliseconds_bucket{${this.getQueryLabels(
+        params,
+      )}}[${range}h])))`,
+    });
+
+    // get p95 latency [range] hours in series of [step]
+    const result = await this.client.queryRange({
```

Comment:
Should we run all those queries in parallel? How fast is it?

Reply:
Locally it's blazing fast, but with larger amounts of load and traffic it might be slower, i will use promise.all